motleycrew/agents/parent.py

```diff
@@ -111,6 +115,27 @@ def compose_prompt(
         Returns:
             The composed prompt.
         """
+        # If input is already a list of messages, return them directly
+        if isinstance(input, list) and all(isinstance(m, BaseMessage) for m in input):
+            if len(self.prompt.input_variables):
+                raise ValueError(
+                    "Cannot use a prompt with variables when input is a list of messages"
+                )
+            output = [HumanMessage(content=self.prompt.format())] + input
```

Comment:
`self.prompt` could be None. I would suggest:
```suggestion
            if self.prompt:
                raise ValueError(
                    "List of messages provided as input, but a prompt is already set for the agent"
                )
```

Reply:
I disagree, the prompt is the generic instruction on what to do, the HumanMessage could for example be an image which changes every time, and the prompt is the instruction on what to do with that image, which makes sense to be in the agent, not in the input - see the change I'm about to commit.